src/awkward/_typetracer.py

```diff
@@ -241,9 +342,19 @@ def from_array(cls: TTypeTracerArray, array, dtype=None) -> TTypeTracerArray:
         if dtype is None:
             dtype = array.dtype
 
-        return cls(dtype, shape=array.shape)
+        if isinstance(array, TypeTracerArray):
+            form_key = array._form_key
+            report = array._report
+        else:
+            form_key = None
+            report = None
+
+        return cls(dtype, array.shape, form_key, report)
+
+    def __init__(self, dtype, shape=None, form_key=None, report=None):
+        self.form_key = form_key
+        self.report = report
 
-    def __init__(self, dtype, shape=None):
         self._dtype = np.dtype(dtype)
         self.shape = shape
```

Comment:
Good point. `shape_touched` isn't very useful if it's always on. If you're assigning a shape to an array (which I think can only happen during construction—any other time it would be a `reshape` call...), then you already know it; you don't need to learn it.

The `touch_shape` call is still in the `shape` property, but not in the setter anymore: e8d39df2e3e51b02544b76b24209b7f311256078.

Reply:
I don't think `__repr__` should count as a touch. It's not something that any ak.Array or even any layout interaction does, and it can't be represented by any Dask nodes. Whether it counts or not, it wouldn't have an impact on any foreseeable applications.